pypeit/core/flat.py

```diff
@@ -28,8 +29,10 @@
 import scipy
 
 
-def fit_flat(flat, mstilts, thismask, slit_left, slit_righ, inmask = None,spec_samp_fine = 1.2,  spec_samp_coarse = 50.0,
-             spat_samp = 5.0, spat_illum_thresh = 0.01, npoly = None, trim_edg = (3.0,3.0),debug = False):
+
+def fit_flat(flat, tilts_dict, thismask_in, slit_left_in, slit_righ_in, inmask = None,spec_samp_fine = 1.2, spec_samp_coarse = 50.0,
```

Comment:
Can you break this function into smaller chunks?  I suggest pulling out the tweaking algorithm into a different function.  You could also pull the debug plots into separate qa plotting functions.

Reply:
I broke our the slit tweaking. I'm leaving the debug/QA stuff where it is because this may end up being subprocess plot in interactive mode. If that is the case, it needs to be defined in the function as with the other interactive subprocess plots. If we dump this out to QA and don't plot to the screen then it could go into a separate function later.